src/compute/src/compute_state.rs

```diff
@@ -907,7 +915,10 @@ impl ReportedFrontier {
 /// State maintained for a compute collection.
 pub struct CollectionState {
     /// Tracks the frontier that has been reported to the controller.
-    pub reported_frontier: ReportedFrontier,
+    ///
+    // TODO(teskje): Now that we explicitly track the collection's `as_of`, we might be able to
+    // simplify this to an `Antichain<Timestamp>` again.
```

Comment:
Can you file an issue for this? Thanks!

Reply:
Done: https://github.com/MaterializeInc/materialize/issues/20766